### ISSUE TITLE
allow profile to limit inputLookupSInopia to Works or to Instances

### ISSUE THREAD

**Issue by jermnelson** [LD4P/sinopia_editor#1153]
ORIGINAL TICKET: Initially we want to support selected typeaheads that add three types of searches, BIBFRAME Works, Instances, and Items. Once [30](ttps://github.com/LD4P/sinopia_indexing_pipeline/issues/30) is fixed, we can support this functionality a couple of different ways:
-  Create specific Sinopia URNs for each of three classes (ie.e BIBFRAME Work urn:ld4p:sinopia:work)
- Use the same `urn:ld4p:sinopia` urn but add a subauthority value for the specific RDF class.

Pros and cons to both approaches.

Blocked by https://github.com/LD4P/sinopia_indexing_pipeline/issues/30

CHALLENGE:
Not all Works will be bf:Work, see notes below. It's also possible that some resources will have type that's a subclass of bf:Work. Would need to take those two possibilities into account.

**Comment by michelleif** [comment]
@CECSpecialistI if you made a template for your Work and gave the template the Resource URI http://rdaregistry.info/Elements/c/C10001, would that achieve what you are trying to do?

**Comment by CECSpecialistI** [comment]
@michelleif we have done this. What we want to be able to do is to mint URI's for things being described using those templates, so that we can refer to them from other entities we describe. It would also be ideal for us to be able to search for these using a type-ahead as is being worked on for BIBFRAME Works, Instances, and Items, but being able to cut and paste minted URI's will work as well if that isn't possible soon.

**Comment by michelleif** [comment]
so if you create a template for a thing, then Sinopia mints a URI. what are you lacking right now as far as minting URIs?
thanks for input on searching, I will add that to existing ticket (or make one) for enhancing the searching / lookup of entities

**Comment by CECSpecialistI** [comment]
Oh, I knew it was supposed to mint a URI, but I can't seem to find verification that URI's are actually being minted. I'm in development right now because we're having trouble getting our profiles loaded into stage (Alexis was talking about that on Slack today and Jeremy is looking into it). I'm trial-cataloging a book and using "Preview RDF" to try and see the URI being minted for a work and I don't see anything. Do I need to publish for a URI to be minted? If so, how do I pull up the saved RDA:Work again if it isn't coded in BIBFRAME?

**Comment by michelleif** [comment]
The URI is minted when you click Save & Publish. A URI is minted only for the main resource described by the template. For example if you have a template with resource class = Work, then we mint a URI for the Work. If you template nests an Instance template, we don't mint a URI for that nested Instance. Right now there is no way to see the work again unless it had a bf:title, so:
What is the RDA property for title? What other fields in your RDA Work description would you want to search on? We could add those to the indexing in this next work cycle.
Do you think there should be a way to just see "recently cataloged stuff" and then you could find your thing regardless of indexing?

**Comment by michelleif** [comment]
@CECSpecialistI what issues are you having when trying to load a profile in stage?

**Comment by CECSpecialistI** [comment]
Same as Alexis, I load the profile, then it isn't available to select from the list.

**Comment by CECSpecialistI** [comment]
We're trying to load the same profile.

**Comment by CECSpecialistI** [comment]
It would be super cool to be able to see "recently cataloged stuff", or to be able to search by URI (we could keep track of those in a pinch). We use http://rdaregistry.info/Elements/w/P10088 for titles of RDA Works, http://rdaregistry.info/Elements/e/P20312 for titles of RDA Expressions, http://rdaregistry.info/Elements/m/P30156 for titles of RDA Manifestations, and we could add a property (http://rdaregistry.info/Elements/i/P40085) for titles of Items if that's the only thing we can use to search for them.

**Comment by CECSpecialistI** [comment]
That is good to know about nested RT's. Is there a plan to change that and begin minting URI's for nested Instances? If not, we might de-nest our Expressions, Manifestations, and Items so we can get URI's for them. Does Sinopia just create a blank node for nested RT's right now?

**Comment by michelleif** [comment]
Yes, currently Sinopia creates blank node for nested RTs. De-nesting would be a good idea for many reasons! on the profiles slack channel there has been some discussion of de-nesting Item and also of de-nesting Instance from Work and vice versa. maybe something that should be discussed in Profiles Affinity Group? Or Sinopia User Group?

**Comment by CECSpecialistI** [comment]
Maybe! It depends...it might be okay for those using the BIBFRAME model since it is so blank-node-y. We're data model outliers so I'm not sure our issues with URI's and searching for different classes/properties would be relevant to the rest of the cohort unless someone is also planning on using our profiles or other experimental non-BIBFRAME ones. I could see issues with having no URI for BF:Instances, though, in a shared cataloging environment.

**Comment by michelleif** [comment]
I'm going to rename this ticket to focus it on the inputLookupSinopia component.
Does https://github.com/LD4P/sinopia_indexing_pipeline/issues/30 mean we already index Sinopia things by type/class?
(The parts of this ticket about searching Sinopia from the Search tab and about a potential browse recent feature are covered in https://github.com/LD4P/sinopia/issues/90)

**Comment by michelleif** [comment]
QUESTION: what would we look up Items by?
QUESTION: if we implement this, we should still have an "all_Sinopia" lookup choice, against everything cataloged in Sinopia

**Comment by michelleif** [comment]
another question: if we implement inputLookupSinopia to have more choices (everything, Works, Instances) then does this mean we get those choices on the Search tab too? Does that come automatically? do we need another ticket for that?

**Comment by michelleif** [comment]
answer to above: on Search tab, faceting can handle this

**Comment by justinlittman** [comment]
Proposed approach is to make this similar to other lookups:
1. Define an authority config for each type. For example:
```
  {
    "label": "Sinopia BIBFRAME instance",
    "uri": "urn:ld4p:sinopia:bibframe:instance",
    "type": "http://id.loc.gov/ontologies/bibframe/Instance"
    "component": "local-lookup"
  },
```
2. Add one or more Sinopia authority configs to a resource template. For example:
```
{
  "id": "test:resource:SinopiaLookup",
  "resourceLabel": "Testing sinopia lookup",
  "remark": "This hits elasticsearch",
  "resourceURI": "http://id.loc.gov/ontologies/bibframe/Instance",
  "propertyTemplates": [
    {
      "propertyURI": "http://id.loc.gov/ontologies/bibframe/instanceOf",
      "propertyLabel": "Instance of (lookup)",
      "remark": "lookup",
      "mandatory": "true",
      "repeatable": "false",
      "type": "lookup",
      "resourceTemplates": [],
      "valueConstraint": {
        "valueTemplateRefs": [],
        "useValuesFrom": [
          "urn:ld4p:sinopia:bibframe:instance",
          "urn:ld4p:sinopia:rda:instance"
        ],
        "valueDataType": {},
        "defaults": []
      }
    }
  ]
}
```
3. When searching, each of the types appears in the dropdown as a separate menu section, like:
![Sinopia](https://user-images.githubusercontent.com/588335/67764903-50531e00-fa21-11e9-87ff-a6f513b2158a.png)

**Comment by jgreben** [comment]
@justinlittman How does (e.g.) `urn:ld4p:sinopia:bibframe:instance` translate to a Sinopia search query?

**Comment by justinlittman** [comment]
This is used to perform a lookup in the authorityConfig, which allows you to get the type "http://id.loc.gov/ontologies/bibframe/Instance". This can then be added to the ES query as a [filter](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/query-dsl-bool-query.html) on the _type_ field.

**Comment by justinlittman** [comment]
And presumably, there might be multiple types.
